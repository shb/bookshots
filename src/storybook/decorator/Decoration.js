import * as styles from './styles'

export default class Badge
{
  constructor (type, props) 
  {
    this.style = styles[type]
    this.props = Object.assign({}, props || {})
  }

  toString ()
  {
    return this.style(this.props)
  }
}