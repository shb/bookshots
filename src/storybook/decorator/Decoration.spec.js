import Decoration from "./Decoration";

describe("Decoration", () =>
{
  it("badge", () => {
    const badge = new Decoration('badge', {
      color: 'red',
      content: "content"
    })
    expect(badge.toString()).toMatchSnapshot()
  })

  it("bullet", () => {
    const badge = new Decoration('bullet', {
      color: 'red'
    })
    expect(badge.toString()).toMatchSnapshot()
  })

})