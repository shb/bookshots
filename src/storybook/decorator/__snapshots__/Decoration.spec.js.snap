// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decoration badge 1`] = `
"{
    display: inline-block;
    content: \\"content\\";
    background: red;
    min-width: 1em;
    border-radius: 10em;
    vertical-align: baseline;
    padding: 2px 1ex 1px;
    font-size: 85%;
    margin: 0 1ex;
    text-align: center;
    font-weight: bold;
    color: white;
    line-height: 1;
    /* position: absolute; */
    right: 2em;
}"
`;

exports[`Decoration bullet 1`] = `
"{
    display: inline-block;
    content: \\"\\";
    background: red;
    width: 1em;
    height: 1em;
    border-radius: 1em;
    font-size: 85%;
    margin: 0 1ex;
    /* position: absolute; */
    right: 2em;
}"
`;
